src/Symfony/Bundle/FrameworkBundle/CacheWarmer/ConfigBuilderCacheWarmer.php

```diff
@@ -68,19 +69,30 @@ public function warmUp(string $cacheDir, ?string $buildDir = null): array
             }
         }
 
+        $configurations = [];
         foreach ($extensions as $extension) {
+            if (null === $configuration = $this->getConfigurationFromExtension($extension)) {
+                continue;
+            }
+
+            $configurations[$extension->getAlias()] = $configuration;
+
             try {
-                $this->dumpExtension($extension, $generator);
+                $generator->build($configurations[$extension->getAlias()]);
             } catch (\Exception $e) {
                 $this->logger?->warning('Failed to generate ConfigBuilder for extension {extensionClass}: '.$e->getMessage(), ['exception' => $e, 'extensionClass' => $extension::class]);
             }
         }
 
+        if (class_exists(ConfigFunctionAwareBuilderGeneratorInterface::class) && $configurations) {
+            $generator->buildConfigFunction($configurations);
```

Comment:
this might miss some tests... ;)
```suggestion
        if ($generator instanceof ConfigFunctionAwareBuilderGeneratorInterface && $configurations) {
            $generator->buildConfigFunction($configurations)();
```

Reply:
Indeed, tests added 👍